### PR TITLE
Isometric draw order

### PR DESCRIPTION
**Pull request by SimonPurdie**
Yo, this looks pretty messy since I was working off a newer version of CleverRaven\Master with your branch merged. I should probably have forked your repo instead?

Let me know if you want me to give the changes in a different way. The relevant stuff is in

cata_tiles.cpp:
    cata_tiles::draw( ... )
    cata_tiles::draw_from_id_string( ... )
    cata_tiles::get_window_tile_counts( ... )

sdl_tiles.cpp
input_context::get_coordinates( ... )

Basically it makes all isotile stuff use the center of the view (e.g. player/cursor) as the 'anchor' for drawing, rather than the topleft corner. The draw order is changed from a naive x/y loop to a screen row/column loop, with row and column being converted to the appropriate x and y for iso / non-iso.

### CODE REVIEW

**Comment by sparr** [comment]
Working from a merged branch instead of my branch means the fastest way I have to actually look at this is to merge it, so I don't have to resolve the merge conflicts between master and my branch independently. So, I'm gonna merge this, and work from there.
